tests/migration_test.go

```diff
@@ -2826,6 +2829,207 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
+			// TODO? Other tests use only 128Mi + the following but my test cluster doesn't do hugepages apparently
+			//migratableVMI.Spec.Domain.Memory = &v1.Memory{
+			//	Hugepages: &v1.Hugepages{PageSize: "2Mi"},
+			//}
+
+			By("creating a template for a pause pod with 2 dedicated CPU cores")
+			pausePod = tests.RenderPod("pause-", nil, nil)
+			pausePod.Spec.Containers[0].Name = "compute"
+			pausePod.Spec.Containers[0].Command = []string{"sleep"}
+			pausePod.Spec.Containers[0].Args = []string{"3600"}
+			pausePod.Spec.Containers[0].Resources = k8sv1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceCPU:    resource.MustParse("2"),
+					k8sv1.ResourceMemory: resource.MustParse("128Mi"),
+				},
+				Limits: k8sv1.ResourceList{
+					k8sv1.ResourceCPU:    resource.MustParse("2"),
+					k8sv1.ResourceMemory: resource.MustParse("128Mi"),
+				},
+			}
+			pausePod.Spec.Affinity = &k8sv1.Affinity{
+				NodeAffinity: &k8sv1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
+						NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
+							{
+								MatchExpressions: []k8sv1.NodeSelectorRequirement{
+									{Key: testLabel2, Operator: k8sv1.NodeSelectorOpIn, Values: []string{"true"}},
+								},
+							},
+						},
+					},
+				},
+			}
+		})
+
+		AfterEach(func() {
+			tests.RemoveLabelFromNode(nodes[0].Name, testLabel1)
+			tests.RemoveLabelFromNode(nodes[1].Name, testLabel2)
+			tests.RemoveLabelFromNode(nodes[1].Name, testLabel1)
+		})
+
+		It("should successfully update a VMI's CPU set on migration", func() {
+			By("ensuring at least 2 worker nodes have cpumanager")
+			Expect(len(nodes)).To(BeNumerically(">=", 2), "at least two worker nodes with cpumanager are required for migration")
+
+			By("starting a VMI on the first node of the list")
+			tests.AddLabelToNode(nodes[0].Name, testLabel1, "true")
+			vmi := tests.CreateVmiOnNodeLabeled(migratableVMI, testLabel1, "true")
+
+			By("waiting until the VirtualMachineInstance starts")
+			tests.WaitForSuccessfulVMIStartWithTimeout(vmi, 120)
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("ensuring the VMI started on the correct node")
+			Expect(vmi.Status.NodeName).To(Equal(nodes[0].Name))
+
+			By("reserving the cores used by the VMI on the second node")
+			var pods []*k8sv1.Pod
+			var pausePodCurrent *k8sv1.Pod
+			tests.AddLabelToNode(nodes[1].Name, testLabel2, "true")
+			for pausePodCurrent = tests.RunPod(pausePod); !hasCommonCores(vmi, pausePodCurrent); pods = append(pods, pausePodCurrent) {
+				By("creating another pod since last didn't have common cores with the VMI")
+				pausePodCurrent = tests.RunPod(pausePod)
+			}
+
+			By("deleting the pods that don't have cores in common with the VMI")
+			for _, pod := range pods {
```

Comment:
won't the additional pod will be in `pod`? 
looking at the for loop, you are always appending the pausePodCurrent, regardless of the hasCommonCores value.

Reply:
You're absolutely right, I got my for loop backwards, I was adding all pods to the list except for the first one, when I want to add all but the last.
It just happened to work because the first pod is usually a winner.... Fixed!